src/iosrv/profile.c

```diff
@@ -0,0 +1,238 @@
+/**
+ * (C) Copyright 2016-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is part of the DAOS server. It implements the DAOS server profile
+ * API.
+ */
+#define D_LOGFAC       DD_FAC(server)
+
+#include <abt.h>
+#include <daos/common.h>
+#include <daos/event.h>
+#include <daos_errno.h>
+#include <daos_srv/bio.h>
+#include <daos_srv/smd.h>
+#include <gurt/list.h>
+#include "drpc_internal.h"
+#include "srv_internal.h"
+
+#define DEFAULT_CHUNK_SIZE  10240
```

Comment:
maybe can define a max  allocated chunk count to limit the memory usage? (when reach the number then give a msg/log and stop allocate new chunk). seems the memory consumption will be large for long running

Reply:
yes, I planed to optimize this in the later patch. but maybe I can do it in this patch.

Reply:
So we consume a slot to record duration each time, this is not a lot because ideally each ULT can handle 100K+ RPC/sec, if we add stats for a few different operations, then 1 million slots can be consumed within a second.

Reply:
any comment on this one?